service/controller/v12/resource/instance/create.go

```diff
@@ -131,6 +162,10 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 	}
 
+	if hasResourceStatus(customObject, Stage, DeploymentCompleted) {
+		return r.handleDeploymentCompletedStatus(ctx, customObject)
```

Comment:
This here is missing a stack trace entry due to missing error masking. Later down I wrote a comment that suggests to actually lift the error handling to this level so that the resource control flow is kept at top level.